docs/content/posts/2023-10-10-release-1.18.0.md

```diff
@@ -0,0 +1,542 @@
+
+### Watermark Alignment and Idleness Detection in SQL
+
+You can now configure [watermark alignment](https://nightlies.apache.org/flink/flink-docs-stable/docs/dev/datastream/event-time/generating_watermarks/#watermark-alignment) 
+and [source idleness timeouts](https://nightlies.apache.org/flink/flink-docs-stable/docs/dev/datastream/event-time/generating_watermarks/#dealing-with-idle-sources) 
+in pure SQL via hints. Previously, these features were only available in the DataStream API.
+
+**More Information**
+- [Documentation](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/queries/hints/)
+- [FLIP-296: Extend watermark-related features for SQL](https://cwiki.apache.org/confluence/display/FLINK/FLIP-296%3A+Extend+watermark-related+features+for+SQL)
+
+## Batch Execution Improvements
+
+### Hybrid Shuffle supports Remote Storage
+
+Hybrid Shuffle supports storing the shuffle data in remote storage. The remote storage path can be configured with the 
+option `taskmanager.network.hybrid-shuffle.remote.path`. Hybrid Shuffle uses less network memory than before by 
+decoupling the memory usage from the number of parallelisms, improving the stability and ease of use. 
+
+For more detailed 
+information, please refer to 
+
+**More Information**
+* [Documentation](https://nightlies.apache.org/flink/flink-docs-master/docs/ops/batch/batch\_shuffle/#hybrid-shuffle)
+* [FLIP-301: Hybrid Shuffle supports Remote Storage](https://cwiki.apache.org/confluence/display/FLINK/FLIP-301%3A+Hybrid+Shuffle+supports+Remote+Storage)
+
+
+### Performance Improvements & TPC-DS Benchmark
+
+In previous releases, the community has done a lot of work to improve batch processing performance, which has led to 
+significant improvements. In this release cycle, community contributors have continued to put a lot of effort into 
+improving batch performance.
+
+#### Runtime Filter for Flink SQL
+
+Runtime filter is a common optimization to improve join performance. It is designed to dynamically generate filter 
+conditions for certain Join queries at runtime to reduce the amount of scanned or shuffled data, avoid unnecessary I/O 
+and network transmission, and speed up the query. We introduced runtime filters in Flink 1.18, and verified its 
+effectiveness through the TPC-DS benchmark, and observed up to 3x speedup for some queries by enabling this feature.
+
+#### Operator Fusion Codegen for Flink SQL
+
+Operator Fusion Codegen improves the execution performance of a query by fusing an operator DAG into a single optimized 
+operator that eliminates virtual function calls, leverages CPU registers for intermediate data and reduces the 
+instruction cache miss. As a general technical optimization, we verified its effectiveness through TPC-DS, and 
+only some batch operators completed fusion codegen support in version 1.18, getting significant performance gains on 
```

Comment:
I am not sure what "only some batch operators completed fusion codegen support in version 1.18" means. Do we need the word "only". I am curious which batch operators completed fusion codegen support in version 1.18.

Reply:
Currently, only three batch operators: Calc, HashAgg, and HashJoin support it.

Reply:
> Currently, only three batch operators: Calc, HashAgg, and HashJoin support it.

It would be good to be explicit here.